core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/parser/YamlSignerParser.java

```diff
@@ -29,18 +31,26 @@
 public class YamlSignerParser implements SignerParser {
   private static final ObjectMapper OBJECT_MAPPER =
       new ObjectMapper(new YAMLFactory()).registerModule(new SigningMetadataModule());
-  final ArtifactSignerFactory artifactSignerFactory;
 
-  public YamlSignerParser(final ArtifactSignerFactory artifactSignerFactory) {
-    this.artifactSignerFactory = artifactSignerFactory;
+  private final BlsArtifactSignerFactory blsFactory;
+  private final Secp256k1ArtifactSignerFactory secpFactory;
+
+  public YamlSignerParser(
+      final BlsArtifactSignerFactory blsFactory, final Secp256k1ArtifactSignerFactory secpFactory) {
+    this.blsFactory = blsFactory;
+    this.secpFactory = secpFactory;
   }
 
   @Override
   public ArtifactSigner parse(final Path metadataPath) {
     try {
       final SigningMetadata metaDataInfo =
           OBJECT_MAPPER.readValue(metadataPath.toFile(), SigningMetadata.class);
-      return metaDataInfo.createSigner(artifactSignerFactory);
+      if (metaDataInfo.getKeyType().equals(KeyType.BLS)) {
+        return metaDataInfo.createSigner(blsFactory);
```

Comment:
nit: Think this would be a bit clearer with a ternary

Reply:
done